### ISSUE TITLE
Demo simulation with custom Navigation2 Behavior Tree

### ISSUE THREAD

**Issue by ljaniec** [neobotix/neo_simulation2#34]
Hello,

Since Neobotix robots in your simulation are used with Nav2 stack, would it be possible to present a task execution example (e.g. picking demo with `wait_for_input` plugin) using custom BT and Nav2 nodes (based on [this](https://navigation.ros.org/tutorials/docs/using_groot.html) tutorial)? With use of `nav2_bt_navigator` or similar.

At the moment I couldn't find any nice and simple example of its use. 
It could be then promoted as a tutorial on ROS Discourse forum or linked on the ROS2/Nav2 website as a Neobotix tutorial.

Best regards
Łukasz Janiec

**Comment by padhupradheep** [comment]
Hey, thanks of the request! Are you sure there is a `wait_for_input` BT Plugin ? or do you just mean `wait` ? Also what do you mean by picking demo? Is it something like dock and toe? If so, at the moment we do not have any plans to develop the BT-Plugins for docking or toeing. 

>  It could be then promoted as a tutorial on ROS Discourse forum or linked on the ROS2/Nav2 website as a Neobotix tutorial.

This indeed is a good idea, we will surely do it, once we develop it.

**Comment by ljaniec** [comment]
You are right, I made a mistake - it is `input_at_waypoint` Waypoint Follower plugin from [there](https://github.com/ros-planning/navigation2/tree/main/nav2_waypoint_follower/plugins). The `wait` BT plugin was from [there](https://navigation.ros.org/plugin_tutorials/docs/writing_new_bt_plugin.html#writing-new-nbt-plugin) and I have mismatched things in my head, I am sorry.

Picking demo - pickup and drop demo, that's it: a robot moves to the point A, wait for the input from the human for package pickup confirmation, moves to the point B, wait for the input ("drop"). It seems like a really good base for more complex stuff later too.

Thanks for your fast answer, I understand that default BT is fine too.

**Comment by padhupradheep** [comment]
> Picking demo - pickup and drop demo, that's it: a robot moves to the point A, wait for the input from the human for package pickup confirmation, moves to the point B, wait for the input ("drop"). It seems like a really good base for more complex stuff later too.

I like this idea, I do not know when will this be added to the BT tree. But I will keep this as an "open" issue, so that I have this in my mind for the next year.

**Comment by ljaniec** [comment]
There:
https://github.com/Adlink-ROS/BT_ros2/
Are some examples with custom BT, but right now the package isn't compiling correctly on my side.

I created an issue with it there: 
https://github.com/Adlink-ROS/neuronbot2/issues/26

It could still be a base for this kind of demo so I wanted to mention it there.
